ophyd/v2/_channelca.py

```diff
@@ -145,10 +136,24 @@ async def put(self, value: T, wait=True):
             timeout=None,
         )
 
+    def _make_ca_descriptor(self, value: AugmentedValue) -> Descriptor:
+        if value.element_count > 1:
+            dtype = "array"
+            shape = [value.element_count]
+        elif isinstance(self._converter, EnumConverter):
+            dtype = "string"
+            shape = []
+            choices = [e.value for e in self._converter.datatype]
+            return dict(source=self.source, dtype=dtype, shape=shape, choices=choices)
+        else:
+            dtype = dbr_to_dtype[value.datatype]
+            shape = []
+        return dict(source=self.source, dtype=dtype, shape=shape)
```

Comment:
I'd like to avoid any `isinstance` checks in the `get_*` functions, please could you add a new `make_descriptor()` function on `CaValueConverter` (and an override in `EnumConverter`) that we can call in `get_descriptor()`?